brave/brave-core
# [Rewards 3.0] Add WDP opt-in to Rewards onboarding

Resolves https://github.com/brave/brave-browser/issues/40041
Resolves https://github.com/brave/brave-browser/issues/40570

<img width="402" alt="Screenshot 2024-10-08 at 10 54 24 AM" src="https://github.com/user-attachments/assets/0b78908b-2b5b-46db-97fe-4d83c37c396c">




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- With the "new Rewards UI" flag enabled,
- Enable Rewards (via the panel or the page).
- Verify that the WDP opt-in screen is shown.
- Verify that user's WDP opt-in status is correctly reflected on `brave://settings/search`
